### PR TITLE
fix(CustomView) : prepared query for host monitoring fixed

### PR DESCRIPTION
**Pull request by Drago37**
### Fixed

* prepared query for host monitoring fixed

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [x] 23.04.x
- [x] 23.10.x
- [x] 24.04.x
- [x] develop

## How this pull request can be tested ?

I do | I get
-- | --
I add a custom view with host monitoring | I can see the hosts in the list of custom view without errors
I execute the unit tests for CentreonLogTest | Unit tests executed successful

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/class/centreonLog.class.php [hunk 1]
```diff
@@ -129,77 +147,379 @@ public static function singleton($uid = 0)
...
+    public function __construct(array $customLogFiles = [], string $pathLogFile = '')
+    {
+        $this->setPathLogFile(empty($pathLogFile) ? _CENTREON_LOG_ : $pathLogFile);
+        // push default logs in log file handler
+        foreach (self::DEFAULT_LOG_FILES as $idLogType => $logFileName) {
+            $this->pushLogFileHandler($idLogType, $logFileName);
+        }
+        // push custom logs in log file handler
+        foreach ($customLogFiles as $idLogType => $logFileName) {
+            $this->pushLogFileHandler($idLogType, $logFileName);
```
Comment (jeremyjaouen) [inline centreon/www/class/centreonLog.class.php:205]:
```suggestion
        foreach (self::DEFAULT_LOG_FILES as $idLogType => $logFileName) {
            $this->pushLogFileHandler($idLogType, $logFileName);
        }
        // push custom logs in log file handler
        foreach ($customLogFiles as $logTypeId => $logFileName) {
            $this->pushLogFileHandler($lodTypeId, $logFileName);
```
Just a suggest to be more convenient with our variable name convetion

**Review by jeremyjaouen — DISMISSED** [review]

File: centreon/www/class/centreonLog.class.php (hunk 1 above)
Comment (Drago37) [reply centreon/www/class/centreonLog.class.php]: fixed :)

**Review by jeremyjaouen — DISMISSED** [review]

File: centreon/www/class/centreonLog.class.php [hunk 2]
```diff
@@ -129,77 +147,379 @@ public static function singleton($uid = 0)
...
+     */
+    public function pushLogFileHandler(int $logTypeId, string $logFileName): CentreonLog
+    {
+        $pathLogFileName = '';
+        $logFile = '';
+        $explodeFineName = explode(DIRECTORY_SEPARATOR, $logFileName);
+        if (! empty($explodeFineName)) {
+            $logFile = $explodeFineName[count($explodeFineName) - 1];
+            unset($explodeFineName[count($explodeFineName) - 1]);
+            $pathLogFileName = implode(DIRECTORY_SEPARATOR, $explodeFineName);
```
Comment (nfragnet) [inline centreon/www/class/centreonLog.class.php:361]: Typo: I guess you meant `explodeFileName`

**Review by nfragnet — COMMENTED** [review]
Nothing to declare about the initial purpose of the PR. That's 👍

I have mixed feelings on the Logger part.

On the one hand it goes into the right direction to improve logs and what we are able to find when something went wrong.

On the other it goes at the opposite of one of our goal which is "reduce code ownership". This PR is more or less rebuilding some tool like Monolog. 

I wonder if it would be possible AND if it would take significantly more time to include a real log tool and stop managing this concern ourselves.

We also need to talk about the backport part

File: centreon/www/class/centreonLog.class.php (hunk 2 above)
Comment (Drago37) [reply centreon/www/class/centreonLog.class.php]: fixed on PR #4862

**Review by a-launois — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

File: centreon/www/widgets/host-monitoring/src/index.php [hunk 3]
```diff
@@ -366,16 +424,26 @@
...
+                AND service_id = 0 ORDER BY entry_time DESC LIMIT 1
+            SQL;
+            $res2 = $dbb->prepare($query);
+            $res2->bindValue(':hostId', $row['host_id'], PDO::PARAM_INT);
+            $res2->execute();
+            if ($row2 = $res2->fetch()) {
+                $data[$row['host_id']]['comment'] = substr($row2['data'], 0, $commentLength);
+            } else {
+                $data[$row['host_id']]['comment'] = '-';
+            }
```
Comment (nfragnet) [inline centreon/www/widgets/host-monitoring/src/index.php:439]: [Nitpick & subjective sharing]

As much as possible I try to :
- have meaningful naming for vars
- avoid if else blocks & replace by ternary format as I find it easier to read.

```suggestion
        $commentRow = $res2->fetch();
        $data[$row['host_id']]['comment'] = $commentRow
            ? substr($commentsRows['data'], 0, $commentLength)
            : '-';
```

Comment (Drago37) [reply centreon/www/widgets/host-monitoring/src/index.php:439]: Yes, you're right! But I only caught the exception to log it, the code is not my production :) I have two revisions, so I won't be making that change. I'm solving this comment.

**Pull request merged**